### DISCUSSION TITLE
Using libpod in C++ project

### DISCUSSION THREAD

**Discussion by remiolivier** [containers/podman#9797] [Q&A] [answered]
Hi,

Is it possible and do you have any pointers/documentation on how to include and use libpod in a C++ project?

I know the REST API is available but I need something with more control and smaller memory consumption. The server adds 100M on my machine when running the podman system service and I ideally need a smaller memory footprint for this limited device.

I found the example in Go and it looks appealing so I started to play with cgo (-buildmode=c-archive) to generate a library (successfully) for linux but I'm missing headers so far to be usable in a C++ project.

If not, is there anyway to control the memory consumption on the server or to optimize?

Disclaimer, I am totally new with Go.

Thank you.

**Comment by TomSweeneyRedHat** [comment, marked as answer]
@remiolivier other than the REST API, there's not a way to call the libpod functions from C++ at the moment, at least as far as I'm aware.  The best way to call the functions from C or C++ would be from a shared library and that was talked about a year ago in this Issue: https://github.com/containers/podman/issues/9325 and I don't think much progress has been made there since.
